sources/zafira-dbaccess/src/main/java/com/qaprosoft/zafira/dbaccess/state/DatabaseStateManagerResolver.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright 2013-2019 Qaprosoft (http://www.qaprosoft.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.qaprosoft.zafira.dbaccess.state;
+
+import com.qaprosoft.zafira.dbaccess.dao.mysql.management.TenancyMapper;
+import com.qaprosoft.zafira.dbaccess.utils.TenancyDataSourceWrapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Component;
+
+@Component
```

Comment:
We need comments for this one so it would be clear why do we need it; that's for sure